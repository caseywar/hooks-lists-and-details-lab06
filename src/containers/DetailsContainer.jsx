import React, { useState, useEffect } from 'react';
import { findVillagerById } from '../services/AnimalCrossingApi';
import VillagerDetail from '../components/villagers/VillagerDetail';

const DetailsContainer = ({match}) => {
    const [loading, setLoading] = useState(true);
    const [villager, setVillager] = useState({});

    useEffect(() => {
        findVillagerById(match.params.id)
        .then(setVillager)
        .finally(() => setLoading(false));
    }, []);

    if (loading) return <h1>Loading...</h1>;

    return (
        <VillagerDetail villager={villager} />
    );
};

export default DetailsContainer;