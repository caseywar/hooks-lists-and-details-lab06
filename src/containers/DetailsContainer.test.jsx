import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import mockVillagerData from '../../fixtures/villager.json';
import DetailsContainer from './AnimalCrossing';
import { MemoryRouter } from 'react-router';


const server = setupServer(
    rest.get('https://ac-vill.herokuapp.com/villagers/5f5fb4bbbfd05c2aed82e460', (req, res, ctx) => {
    return res(ctx.json(mockVillagerData));
    })
);

describe('Details container', () => {
    beforeAll(() => server.listen());
    afterAll(() => server.close());

    it.skip('displays the loading screen', async () => {
        render(
            <MemoryRouter>
                <DetailsContainer 
        match={{
            params: {
                id: '5f5fb4bbbfd05c2aed82e460'
            }
        }}
        />
            </MemoryRouter>
);

        screen.getByText('Loading...');

        const villager = screen.findByTestId('villager');
        waitFor(() => expect(villager).toMatchSnapshot());
    })
})