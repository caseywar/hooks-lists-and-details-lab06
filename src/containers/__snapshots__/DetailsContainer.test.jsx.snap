// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details container displays the loading screen 1`] = `Promise {}`;
